controllers/review.controller.js

```diff
@@ -29,27 +30,54 @@ class ReviewController {
         return this.formatReview(review);
     }
 
-    create(author, movieId, review) {
-        return this.reviewRepository.create({
+    async updateMovieRating(movieId) {
+        const reviews = await this.getAllByMovieId(movieId);
+        const weight = 1 / reviews.length;
+        const stats = reviews.reduce((statistics, r) => {
+            statistics.count += 1;
+            statistics.average += r.rating * weight;
+            statistics.distribution[r.rating - 1] += 1;
```

Comment:
The distribution function shifts the rating into their respective slot?

Reply:
This line increments the appropriate counter for that number of stars eg. for a 5 star rating, we will increase the counter at position 4 by 1.